### PR TITLE
Feature/game parameters UI

### PR DESCRIPTION
**Pull request by gauthierboeshertz**

### CODE REVIEW

File: app/src/main/java/sdp/moneyrun/database/GameDbData.java [hunk 1]
```diff
@@ -60,6 +63,56 @@ public GameDbData(String name,
...
+        if(name == null){
+            throw new IllegalArgumentException("name should not be null.");
+        }
+        if(host == null){
+            throw new IllegalArgumentException("host should not be null.");
+        }
+        if(players == null){
+            throw new IllegalArgumentException("players should not be null.");
+        }
+        if(startLocation == null){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/database/GameDbData.java:85]: Easy test case that can improve coverage

File: app/src/main/java/sdp/moneyrun/game/Game.java [hunk 2]
```diff
@@ -90,6 +90,56 @@ public Game(String name,
...
+            throw new IllegalArgumentException("Number of coins should be bigger than 0.");
+        }
+        if (radius <=0 ) {
+            throw new IllegalArgumentException("Radius should be bigger than 0.");
+        }
+        if (duration <=0) {
+            throw new IllegalArgumentException("Duration should be bigger than 0.");
+        }
+
+        this.id = null;
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/game/Game.java:132]: I think initializing a variable to null is useless, if a variable is not initialized it is already null.

File: app/src/main/java/sdp/moneyrun/game/Game.java [hunk 3]
```diff
@@ -136,6 +186,54 @@ public Game(String name,
...
+        }
+        if (radius <=0 ) {
+            throw new IllegalArgumentException("Radius should be bigger than 0.");
+        }
+        if (duration <=0) {
+            throw new IllegalArgumentException("Duration should be bigger than 0.");
+        }
+
+
+        this.id = null;
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/game/Game.java:229]: Same here for the id and you might want to test this constructor

File: app/src/main/res/layout/new_game_popup.xml [hunk 4]
```diff
@@ -37,19 +40,44 @@
...
                 android:ems="10"
                 android:hint="Enter max player count"
                 android:inputType="number" />
 
+            <EditText
+                android:id="@+id/newGameNumCoins"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:ems="10"
+                android:hint="Enter the number of coins "
```
Comment (Alvaro2112) [inline app/src/main/res/layout/new_game_popup.xml:61]: You should use resources for the messages you use.

File: app/src/main/res/layout/new_game_popup.xml [hunk 5]
```diff
@@ -37,19 +40,44 @@
...
+            <EditText
+                android:id="@+id/newGameDuration"
+                android:layout_width="match_parent"
+                android:layout_height="78dp"
+                android:ems="10"
+                android:hint="Enter the duration of the game in minutes"
+                android:inputType="numberDecimal" />
+
+            <EditText
+                android:id="@+id/newGameRadius"
```
Comment (Alvaro2112) [inline app/src/main/res/layout/new_game_popup.xml:73]: You are missing the autofill hints attribute here

**Review by Alvaro2112 — CHANGES_REQUESTED** [review]
You should try to fix the CC issues and the warnings from Cirrus, also do not forget to reformat the code, everything else looks good though.

File: app/src/main/java/sdp/moneyrun/database/GameDbData.java (hunk 1 above)
Comment (gauthierboeshertz) [reply app/src/main/java/sdp/moneyrun/database/GameDbData.java:85]: Yes

File: app/src/main/java/sdp/moneyrun/game/Game.java (hunk 2 above)
Comment (gauthierboeshertz) [reply app/src/main/java/sdp/moneyrun/game/Game.java]: Yes

File: app/src/main/java/sdp/moneyrun/game/Game.java (hunk 3 above)
Comment (gauthierboeshertz) [reply app/src/main/java/sdp/moneyrun/game/Game.java]: Yes

File: app/src/main/res/layout/new_game_popup.xml (hunk 4 above)
Comment (gauthierboeshertz) [reply app/src/main/res/layout/new_game_popup.xml]: Yes

File: app/src/main/res/layout/new_game_popup.xml (hunk 5 above)
Comment (gauthierboeshertz) [reply app/src/main/res/layout/new_game_popup.xml:77]: Yes

**Review by Alvaro2112 — APPROVED** [review]

**Review by Tesa27 — APPROVED** [review]
Looks good!

**Pull request merged**